clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -150,19 +209,19 @@ public void testStartupAndTearDown() throws InterruptedException {
     }
 
     @Test
-    public void testApplicationEvent() {
-        ApplicationEvent e = new PollEvent(100);
-        applicationEventsQueue.add(e);
+    void testRequestManagersArePolledOnce() {
         consumerNetworkThread.runOnce();
-        verify(applicationEventProcessor, times(1)).process(e);
+        requestManagers.entries().forEach(rmo -> rmo.ifPresent(rm -> verify(rm).poll(anyLong())));
+        requestManagers.entries().forEach(rmo -> rmo.ifPresent(rm -> verify(rm).maximumTimeToWait(anyLong())));
+        verify(networkClientDelegate).poll(anyLong(), anyLong());
     }
 
     @Test
-    public void testMetadataUpdateEvent() {
-        ApplicationEvent e = new NewTopicsMetadataUpdateRequestEvent();
+    public void testApplicationEvent() {
```

Comment:
Not introduced by this PR, but reviewing this one I noticed that we have lots of similar tests for checking that runOnce processes the events (ex. testResetPositionsEventIsProcessed, testSyncCommitEvent, testAsyncCommitEvent...), all doing the same (add event, run once, check is processed). Could we parametrize this and have a single test maybe? (same shape as this one but receiving the event as param). Should allow us to remove lots that, in the end this component does not care about the specific events, it's just responsible for processing whatever event is added.

Reply:
+1

Reply:
I tried this out, it seems like parameterizing the tests into one will not work, it keeps giving me an error saying that the parameters must be constant. I did some research, it looks like it can be done, but would require an enum class to get it to work. Not sure if this is something we would want to do just for these few tests, but let me know.

Reply:
I had something like this in mind:
```
    @ParameterizedTest
    @MethodSource("appEvents")
    public void testEventIsProcessed(ApplicationEvent e) {
        applicationEventsQueue.add(e);
        consumerNetworkThread.runOnce();
        verify(applicationEventProcessor).process(any(e.getClass()));
        assertTrue(applicationEventsQueue.isEmpty());
    }

    private static Stream<Arguments> appEvents() {
        return Stream.of(
            Arguments.of(new PollEvent(100)),
            Arguments.of(new NewTopicsMetadataUpdateRequestEvent()));
    }
```
(extending `appEvents` with all the event instances that are individually created in each testxxxIsProcessed so we can replace them all with this single parametrized test)